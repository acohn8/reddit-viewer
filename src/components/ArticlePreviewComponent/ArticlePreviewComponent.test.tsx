import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';

import ArticlePreview from './ArticlePreviewComponent';

describe('ArticlePreviewComponent', () => {
  const output = shallow(<ArticlePreview
    link="www.newyorker.com"
    source="New Yorker"
    text="a cool article2"
    title="breaking"
    image="image.jpg"
    postDate={new Date(1536423624 * 1000)}
    icon="newspaper outline"
    upVotes={7}
    comments={5}
  />);
  it('should render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should have a source', () => {
    expect(output.find('New Yorker'));
  });
  it('should have a title', () => {
    expect(output.find('breaking'));
  });
  it('should have a icon', () => {
    expect(output.find('newspaper outline'));
  });
  it('should have  text', () => {
    expect(output.find('a cool article2'));
  });
  it('should have an image', () => {
    expect(output.find('image.jpg'));
  });
});
