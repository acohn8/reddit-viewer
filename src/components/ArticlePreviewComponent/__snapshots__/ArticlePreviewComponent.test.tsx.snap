// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePreviewComponent should render correctly 1`] = `
<FeedEvent>
  <FeedContent>
    <Header
      as="a"
      content="breaking"
      href="www.newyorker.com"
      icon="newspaper outline"
      size="medium"
      subheader="a day ago"
      target="_blank"
    />
    <FeedSummary>
      New Yorker
    </FeedSummary>
    <FeedExtra
      style={
        Object {
          "margin": "auto",
          "maxWidth": 400,
        }
      }
    >
      <Image
        as="a"
        fluid={true}
        href="www.newyorker.com"
        src="image.jpg"
        target="_blank"
        ui={true}
      />
    </FeedExtra>
    <FeedMeta>
      <FeedLike
        as="a"
      >
        <Icon
          as="i"
          name="like"
        />
        7
      </FeedLike>
      <FeedLike
        as="a"
        onClick={[Function]}
      >
        <PostView
          image="image.jpg"
          title="breaking"
        >
          5
        </PostView>
      </FeedLike>
    </FeedMeta>
  </FeedContent>
</FeedEvent>
`;
